features/support/commit_helpers.rb

```diff
@@ -138,5 +144,12 @@ def verify_commits commits_array
 
   actual_commits = commits_in_repo
 
+  # Leave only the expected keys in actual_commits
+  unless expected_commits[0].key? :file_content
+    actual_commits.each do |commit_data|
+      commit_data.delete :file_content
+    end
+  end
```

Comment:
Then we silently add expectations that haven't been provided by the spec. I don't like that. It feels brittle, and might do unexpected things in edge cases. It's also slower. I'd rather really only verify what is explicitly provided.